source/common/upstream/subset_lb.cc

```diff
@@ -602,11 +603,15 @@ void SubsetLoadBalancer::purgeEmptySubsets(LbSubsetMap& subsets) {
         stats_.lb_subsets_removed_.inc();
       }
 
-      it = subset_it->second.erase(it);
+      auto next_it = std::next(it);
+      subset_it->second.erase(it);
+      it = next_it;
```

Comment:
It may be worth doing a separate PR for non-trivial transformations like this one that require use of std::next to work around the erase return value issue.

Reply:
There is an erase that takes a range like the stdlib equivalents and returns an iterator which I considered, though it seems worse: https://github.com/abseil/abseil-cpp/blob/38db52adb2eabc0969195b33b30763e0a1285ef9/absl/container/node_hash_map.h#L221
- we would have to use `std::next` to get the end of the range anyways
- the erase method would still do some iteration/extra comparisons even if we're moving the iterator one position in practice
- it seems absl moved to this signature to guarantee an O(1) erase: https://abseil.io/docs/cpp/guides/container#hash-tables `Guarantees an O(1) erase method by returning void instead of an iterator.`

We can pull these related changes into a separate PR

Reply:
After thinking about it some more, I think that the change you're making is equivalent to the original, idiomatic and, a trivial enough refactor.  Keeping it in this PR is fine.